### PR TITLE
SameDiff fixes and naming

### PR DESCRIPTION
**Pull request by rnett**
* Remove `SDVariable` inplace ops
 * Easy TF import ops (`importFrozenTF`, no tests, should I?)
 * Naming centralized to `SameDiff.generateNewVarName` and `getOpName`, updated to always use `<base>_#:#` form.
 * SDVariable ops without a name parameter now use null instead of the op name constant (same resulting names, but removes some issues).
 * Fixed some NPEs from `OpValidation` 
 * `updateVariableNameAndReference` now doesn't do anything if the name passed in is null, and the variable with the existing name is the variable to update ([here](https://github.com/SkymindIO/deeplearning4j/blob/675fd3bf7575954ad79306e1b589b194acfa9c5f/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java#L4379-L4379)).

### CODE REVIEW

**Comment by rnett** [comment]
Also added `SDVariable.eval` with placeholders argument.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java [hunk 1]
```diff
@@ -5680,4 +5655,139 @@ public String newBlockName(String baseName){
...
+        // if we already have a name like "op_2", start from trying "op_3"
+        if(base.contains("_")){
+            String lastPart = base.substring(base.lastIndexOf('_') + 1);
+            boolean isInt;
+            try{
+                start = Integer.parseInt(lastPart) + 1;
+                isInt = true;
+            } catch (NumberFormatException e){
+                isInt = false;
+            }
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java]: Regex would be cleaner: `boolean isInt = base.matches(".*_\\d+")`

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java [hunk 2]
```diff
@@ -5680,4 +5655,139 @@ public String newBlockName(String baseName){
...
+
+        if(argIndex > 0 && base.contains(":")){
+            String lastPart = base.substring(base.lastIndexOf(':') + 1);
+            boolean isInt;
+            try{
+                argIndex = Integer.parseInt(lastPart) + 1;
+                isInt = true;
+            } catch (NumberFormatException e){
+                isInt = false;
+            }
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java]: Regex as above

**Comment by rnett** [comment]
Should probably re-run `OpValidation` with the NPE fixes, I'm not sure how to get the test case coverage numbers (if I just run `OpValidationSuite` they are all 0) or I would.

**Pull request merged**